service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -171,6 +180,40 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
     // Fetch existing method:
     MethodVersion methodVersion = methodVersionDao.getMethodVersion(request.getMethodVersionId());
 
+    // convert method url to raw GitHub url and use that while calling Cromwell's submit workflow
+    // endpoint
+    String rawMethodUrl;
+    try {
+      rawMethodUrl =
+          MethodUtil.convertToRawGithubUrl(
+              methodVersion.url(),
+              Objects.requireNonNull(
+                  PostMethodRequest.MethodSourceEnum.fromValue(
+                      methodVersion.method().methodSource())),
+              methodVersion.name(),
+              dockstoreService);
+
+      // this could happen if there was no url or empty url received in the Dockstore workflow's
+      // descriptor response
+      if (rawMethodUrl == null || rawMethodUrl.isEmpty()) {
+        String errorMsg =
+            "Error while retrieving WDL url for Dockstore workflow. No workflow url found specified path.";
+        log.error(errorMsg);
+        return new ResponseEntity<>(
+            new RunSetStateResponse().errors(errorMsg), HttpStatus.BAD_REQUEST);
+      }
+    } catch (URISyntaxException
+        | MalformedURLException
+        | bio.terra.dockstore.client.ApiException e) {
+      // the flow shouldn't reach here since if it was invalid URL it should have been caught in
+      // when method was imported
+      String errorMsg =
+          "Something went wrong while submitting workflow. Error: %s".formatted(e.getMessage());
+      log.error(errorMsg, e);
+      return new ResponseEntity<>(
+          new RunSetStateResponse().errors(errorMsg), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
```

Comment:
And again here

Reply:
Although... given the "should have been caught in when method was imported" comment, maybe we _DO_ want to make this a more strong type or error log...

Reply:
I would prefer to leave this as `log.error` so that we are alerted about the flow reaching this point?